### ISSUE TITLE
`push()` and `replace()` will keep current `search` and `hash` string

### ISSUE THREAD

**Issue by yangmingshan** [ReactTraining/history#814]
**When**
Current location is `http://localhost?a=a#b`

**Do**
```js
history.push('/');
// OR
history.replace('/');
```

**Expect**
Location change to `http://localhost`

**Actual**
Location still is `http://localhost?a=a#b`

**Comment by arnaudhuc** [comment]
Got the same issue

**Comment by jnields** [comment]
Yeah, this definitely seems like a bug to me too as it's a breaking change that is not documented with the 5.0.0 release, and just feels very unintuitive.

https://github.com/ReactTraining/history/releases/tag/v5.0.0

**Comment by GordonSmith** [comment]
I think what I am seeing the same issue, but with "createHashHistory":

1. My URL is:  `http://localhost:8080/esp/files/index.html#/main?lastNDays=31&protected=false&state=completed`
2. Use edits the URL in the address bar to:  `http://localhost:8080/esp/files/index.html#/main?lastNDays=999&protected=true`
3. In `history.listen` I still get:  `?lastNDays=31&protected=false&state=completed` for the search...

**Comment by Hypnosphi** [comment]
Here's my workaround:

```js
const push = (to, state) => history.push({
  hash: '',
  search: '',
  ...(typeof to === 'string' ? parsePath(to) : to)
}, state)
```

**Comment by zhangdingding** [comment]
It looks more like a bug than a feature 🤔

**Comment by unbugx** [comment]
same is here #808
